doc/sonic-vrf-hld-v1.0.md

```diff
@@ -81,10 +86,48 @@ The fallback feature which defined by RFC4364 is very useful for specified VRF u
 
 7. VRF Scalability: Currently VRF number can be supported up to 1000 after fixing a bug in FRR.
 
-In this release, supporting requirement 5) and 6) are not supported. See next section for details.
+In this release, requirement 5) is not supported. See next section for details.
 
 Note: linux kernel use VRF master device to support VRF and it supports admin up/down on VRF master device. But we don't plan to support VRF level up/down state on SONIC.
 
+## Functionality
+### Target Deployment Use Cases
+Virtual Routing and Forwarding is used to achieve network virtualization and traffic separation over on a shared network infrastructure in Enterprise and DC delpoyments.
+
+![Deployment use case](https://github.com/preetham-singh/SONiC-1/blob/master/images/vrf_hld/Multi-VRF_Deployment.png "Figure 1: Multi VRF Deployment use case")
+__Figure 1: Multi VRF Deployment use case__
+
+Above figure depicts typical customer deployment where multiple Customer facing devices are connected to Provider edge routers.
+With such deployment Provider Edge routers associate each input interface with one VRF instance.
+Note that multiple input interfaces can be associated to a VRF instance. This input interface can be Physical interface, Port-channel or Ve interface.
+
+### Functional Description
+
+Multi-VRF is the ability to maintain multiple "Virtual Routing and Forwarding"(VRF) tables on the same Provider Edge router.
+Multi-VRF uses multiple instances of a routing protocol such as BGP to exchange routing information among peer provider edge routers.
+The Multi-VRF capable Provider Edge router maps an input customer interface to a unique VRF instance. Provider Edge router maintains unique VRF table for each VRF instance on that Provider Edge router.
+
+Multi-VRF routers communicate with one another by exchanging route information in the VRF table with the neighboring Provider Edge router.
+This exchange of information among the Provider Edge routers is done using routing protocol like BGP.
+Customers connect to Provider Edge routers in the network using Customer Edge routers as shown in Figure 1. 
+
+Due to this overlapping address spaces can be maintained among the different VRF instances.
+
+FRR's BGP implementation is capable of running multiple autonomous systems (AS) at once. Each configured AS is associated with a VRF instance. The  SONiC VRF implementation will enable this capability of FRR BGP in SONiC.
```

Comment:
Please describe the use case for this. It seems like this would be to put different VRFs under different administrative control, which seems like an unlikely use case for SONiC, no?